packages/core/data-transfer/src/strapi/queries/link.ts

```diff
@@ -171,6 +171,36 @@ export const createLinkQuery = (strapi: LoadedStrapi, trx?: Knex.Transaction) =>
           yield link;
         }
       }
+
+      if (attribute.morphColumn) {
+        const { typeColumn, idColumn } = attribute.morphColumn;
+
+        const qb = connection
+          .queryBuilder()
+          .select('id', typeColumn.name, idColumn.name)
+          .from(addSchema(metadata.tableName))
+          .whereNotNull(typeColumn.name)
+          .whereNotNull(idColumn.name);
+
+        if (trx) {
+          qb.transacting(trx);
+        }
+
+        const entries = await qb;
+
+        for (const entry of entries) {
+          const ref = entry[idColumn.name];
+
+          if (ref !== null) {
```

Comment:
Technically `ref` cannot be `null` here since we're excluding this possibility in the where clauses.

I think you can remove this `if`